### DISCUSSION TITLE
Output to framebuffer

### DISCUSSION THREAD

**Discussion by piranna** [servo/servo#10385] [General]
Can `servo` be compiled as a statically linked binary to use the Linux framebuffer as graphical output without any other external dependencies?

**Comment by nox** [comment]
Is this something we want to do? Cc @pcwalton @glennw

**Comment by emilio** [comment]
I don't think this is an easy task at all. I think we have an OSMesa back-end to allow for software rendering, and could be used to get a framebuffer provided you build osmesa-src.

**Comment by SimonSapin** [comment]
Software rendering is one thing, but I also notice this:

> a statically linked binary […] without any other external dependencies

Servo has many C and C++ dependencies with each their own build system and recursive dependencies. Many of those are dynamically linked, and almost all (on a "mainstream" distro) dynamically link to glibc.

Porting all of this to fully static linking and musl libc would take a very considerable amount of work, for questionable benefits IMO. Even if somebody volunteered to do that work, I’m not sure we’d want to maintain it.

As such I’m gonna close.

**Comment by piranna** [comment]
That doesn't answer the main question about rendering to framebuffer, please re-open.
